lib/src/Uncertainty/Algorithm/MetaModel/Kriging/KrigingEvaluation.cxx

```diff
@@ -141,14 +141,58 @@ struct KrigingEvaluationPointFunctor
 
 }; // struct KrigingEvaluationPointFunctor
 
+struct KrigingEvaluationPointFunctor1D
+{
+  const Point & input_;
+  const KrigingEvaluation & evaluation_;
+  Scalar accumulator_;
+
+  KrigingEvaluationPointFunctor1D(const Point & input,
+                                  const KrigingEvaluation & evaluation)
+    : input_(input)
+    , evaluation_(evaluation)
+    , accumulator_(evaluation.getOutputDimension())
+  {}
+
+  KrigingEvaluationPointFunctor1D(const KrigingEvaluationPointFunctor1D & other,
+                                  TBB::Split)
```

Comment:
here

Reply:
I see. It is useless but it is a copy of `KrigingEvaluationPointFunctor`, which has the same actors ;-)

Reply:
At this point it looks quite good. With master we have:
![perf_master](https://user-images.githubusercontent.com/8220933/78187981-bed45600-746f-11ea-83a4-a34aa0e1031a.png)

We see that 2/3 of the time spent in KrigingEvaluation::operator() is spent in CovarianceModelImplementation::operator() and 1/3 is spent in the creation/destruction of temporary Point. On the benchmark I get 370 evals/s

By replacing the call to CovarianceModelImplementation::operator() by a call to CovarianceModelImplementation::computeAsScalar(), we avoid a costly multiplication by a 1x1 matrix and also the creation of some Point to transfer arguments and values:
![perf_regis](https://user-images.githubusercontent.com/8220933/78188688-f263b000-7470-11ea-9edf-15f0a75a7b8c.png)

Now, half of the time is spent computing exp(), which is much better. On the benchmark I get 761 evals/s. But now the time spent in the creation/destruction of Point, even if lower than before, represents half of the total time.

Thanks to Sofiane's improvement, using iterators instead of duplicating the data, we completely remove this part:
![perf_sofiane](https://user-images.githubusercontent.com/8220933/78188834-35be1e80-7471-11ea-9cc3-a3ced1bc02cb.png)

Now, approximately 90% of the time is spent in computeAsScalar(), in which 2/3 of the time is spent computing exp(). On the benchmark I get 1570 evals/s. The final speed-up is 4.24x, not so bad for the very common case of scalar output!